src/merkle/store.rs

```diff
@@ -110,20 +110,39 @@ impl MerkleStore {
 
     /// Returns the path for the node at `index` rooted on the tree `root`.
     ///
-    /// The path starts at the sibling of the target leaf.
-    ///
     /// # Errors
     ///
     /// This will return `NodeNotInStorage` if the element is not present in the store.
-    pub fn get_path(&self, root: Word, mut index: NodeIndex) -> Result<MerklePath, MerkleError> {
-        let mut path = Vec::with_capacity(index.depth().saturating_sub(1) as usize);
-        while index.depth() > 0 {
-            let sibling = index.sibling();
-            index.move_up();
-            let node = self.get_node(root, sibling)?;
-            path.push(node);
+    pub fn get_path(
+        &self,
+        root: Word,
+        index: NodeIndex,
+    ) -> Result<(Word, MerklePath), MerkleError> {
+        let mut hash: RpoDigest = root.into();
+        let mut path = Vec::new();
```

Comment:
optional: this vec can be created `with_capacity` `index.depth()`

Reply:
done